lightning/src/routing/router.rs

```diff
@@ -603,6 +604,17 @@ impl RouteParameters {
 	pub fn from_payment_params_and_value(payment_params: PaymentParameters, final_value_msat: u64) -> Self {
 		Self { payment_params, final_value_msat, max_total_routing_fee_msat: Some(final_value_msat / 100 + 50_000) }
 	}
+
+	/// Sets the maximum number of hops that can be included in a payment path, based on the provided
+	/// [`RecipientOnionFields`] and blinded paths.
+	pub fn set_max_path_length(
```

Comment:
Why do we expose this if `max_path_length` is overridden by `outbound_payments.rs`?

Reply:
Hmmm... I see, though we should update `ChannelManager::send_payment`'s docs to indicate that `max_path_length` is not honored then. Or maybe it would it be better to use the existing setting if it is smaller than what is computed instead of the default `MAX_PATH_LENGTH_ESTIMATE`?

Reply:
> Or maybe it would it be better to use the existing setting if it is smaller than what is computed instead of the default MAX_PATH_LENGTH_ESTIMATE?

I think that's a good point, went this way in a fixup.